models/modules/resnet_architecture/mobile_resnet_generator.py

```diff
@@ -321,18 +323,23 @@ def forward(self, input, extract_layer_ids=[], encode_only=False):
 
         attentions =[]
         
-        for i in range(self.nb_attn):
+        for i in range(self.nb_mask_attn):
             attentions.append(attention[:, i:i+1, :, :].repeat(1, 3, 1, 1))
 
         outputs = []
         
-        for i in range(self.nb_attn-self.nb_mask_input):
+        for i in range(self.nb_mask_attn-self.nb_mask_input):
             outputs.append(images[i]*attentions[i])
-        for i in range(self.nb_attn-self.nb_mask_input,self.nb_attn):
+        for i in range(self.nb_mask_attn-self.nb_mask_input,self.nb_mask_attn):
             outputs.append(input * attentions[i])
+
+        if get_attention_masks:
```

Comment:
Make an other function to output everything, `forward` calls this other function and compute the final image